libs/api/libbpf_map.cpp

```diff
@@ -28,38 +28,13 @@ bpf_map__prev(const struct bpf_map* next, const struct bpf_object* object)
 int
 bpf_map__unpin(struct bpf_map* map, const char* path)
 {
-    int err;
-
-    if (map == NULL) {
-        return libbpf_err(-EINVAL);
-    }
-
-    err = ebpf_api_unpin_object((const uint8_t*)path, (uint32_t)strlen(path));
-    if (err) {
-        return libbpf_err(err);
-    }
-
-    map->pinned = false;
-
-    return 0;
+    return libbpf_err(ebpf_map_unpin(map, path));
```

Comment:
```suggestion
    return libbpf_err(-ebpf_map_unpin(map, path));
```
ebpf_map_unpin should return ebpf_result_t which needs to be turned negative to become a libbpf error